composeApp/src/commonMain/kotlin/sp/bvantur/tasky/core/ui/components/PasswordTextField.kt

```diff
@@ -0,0 +1,66 @@
+package sp.bvantur.tasky.core.ui.components
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Visibility
+import androidx.compose.material.icons.filled.VisibilityOff
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.OutlinedTextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.PasswordVisualTransformation
+import androidx.compose.ui.text.input.VisualTransformation
+import org.jetbrains.compose.resources.stringResource
+import tasky.composeapp.generated.resources.Res
+import tasky.composeapp.generated.resources.password_visibility_icon
+
+@Composable
+fun PasswordTextField(
+    value: String,
+    placeholder: String,
+    onValueChange: (String) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val showPassword = rememberSaveable { mutableStateOf(false) }
```

Comment:
In this specific case, I would even argue that `rememberSaveable` is NOT what we want, since it possibly reveals sensitive information. Imagine the user has their password visible and then leaves the app, which then gets killed by the system. Coming back to this screen would immediately reveal their password since this boolean state is restored (they might have someone else watching their phone and wouldn't expect this). Quite a rare scenario, but if it happens, wouldn't be that good in times where people use the same password across the internet 😄

Reply:
Good points. I will change it from rememberSaveable to remember. I would still want to keep it within this specific component to not complicate usage of it by demanding handling of password visibility from the outside. In the end, this is still implemented as behavior of the UI component and it doesn't make any impact on the outside in which state it is.